lib/cfilters.h

```diff
@@ -27,11 +27,25 @@
 
 struct Curl_cfilter;
 struct Curl_easy;
+struct Curl_dns_entry;
+struct connectdata;
 
-/* Destroy a filter instance. Implementations MUST NOT chain calls to cf->next.
+/**
+ * Macro intended for DEBUGF logging, use like:
+ * DEBUGF(infof(data, CFLOG(cf, "this filter %s rocks"), "very much"));
+ * and it will output:
+ * [CF-1-0-SSL] this filter very much rocks
+ * on connection #1 with sockindex 0 for filter of type "SSL".
  */
-typedef void     Curl_cf_destroy(struct Curl_cfilter *cf,
-                                 struct Curl_easy *data);
+#define CFLOG(cf,msg)  \
+  "[CF-%ld-%d-%s] "msg, (cf)->conn->connection_id, \
+  (cf)->sockindex, (cf)->cft->name
```

Comment:
In the vtls filters, yes. I'f like to get it into all of them. This might be a matter of taste, but I found such things useful in analyzing/filtering log files.

Reply:
I propose sth like this:

```
CLOG(c,sockindex,msg) \
 "[CONN-%ld-%d] "msg, (conn)->connection_id, sockindex, 
CFLOG(cf,msg)\
   "[CONN-%ld-%d][CF-%s] "msg, (cf)->conn->connection_id, \
  (cf)->sockindex, (cf)->cft->name
```

We curently have no Curl_easy ID, I think, otherwise it might be useful too, esp for H2/H3 connections.

Reply:
Extended macros and added them to urldata.h in recent push.